### PR TITLE
config merge operation for iOS devices not clear if rollback was successful

### PR DESCRIPTION
**Pull request by network-shark**
When the config merge operation fails napalm quits with  : "Configuration merge failed; automatic rollback attempted " But it does not tell you, if the rollback was successful or failed. Now it does .

I hope you find it useful.

### CODE REVIEW

File: napalm/ios/ios.py [hunk 1]
```diff
@@ -581,14 +593,16 @@ def rollback(self):
...
+        output = self._commit_handler(cmd)
 
         # After a rollback - we no longer know whether this is configured or not.
         self.prompt_quiet_configured = None
 
         # Save config to startup
         self.device.save_config()
 
+        return output
+
```
Comment (ktbyers) [inline napalm/ios/ios.py:605]: Why was this changed to return `output` from _commit_handler? That didn't seem necessary for the PR?

Comment (ktbyers) [reply napalm/ios/ios.py:605]: @network-shark

Comment (mirceaulinic) [reply napalm/ios/ios.py:605]: Bump @network-shark

Comment (ktbyers) [reply napalm/ios/ios.py:605]: Okay, this was my misunderstanding...I don't think we can change this without changing the behavior of all the drivers, however (i.e. we would be changing what the `rollback` method returns).

Probably an easier solution is just to key off whether `rollback()` generates an exception or not (i.e. if `rollback` does not generate an exception then rollback worked...anything else should be a bug.

Comment (mirceaulinic) [reply napalm/ios/ios.py:605]: @network-shark did you want to reply to Kirk's suggestion above?

**Review by network-shark — COMMENTED** [review]
```
if "Invalid input detected" in output:
              err_header = "Configuration merge failed; automatic rollback attempted"
              merge_error = "{0}:\n{1}".format(err_header, output)

              try:
                  self.rollback()
              except:
                  raise
              finally:
                  raise MergeConfigException(merge_error)
``` 
Like this ?

File: napalm/ios/ios.py (hunk 1 above)
Comment (network-shark) [reply napalm/ios/ios.py:605]:
```
if "Invalid input detected" in output:
              err_header = "Configuration merge failed; automatic rollback attempted"
              merge_error = "{0}:\n{1}".format(err_header, output)

              try:
                  self.rollback()
              except:
                  raise
              else:
                  raise MergeConfigException(merge_error)
``` 
Like this ? @mirceaulinic  @ktbyers

Comment (ktbyers) [reply napalm/ios/ios.py:605]: Probably something more like:

```python
    def rollback(self):
        """Rollback configuration to filename or to self.rollback_cfg file."""
        filename = self.rollback_cfg
        cfg_file = self._gen_full_path(filename)
        if not self._check_file_exists(cfg_file):
            raise ReplaceConfigException("Rollback config file does not exist")
        cmd = "configure replace {} force".format(cfg_file)
        self._commit_handler(cmd)
+      output = self._commit_handler(cmd)

        # After a rollback - we no longer know whether this is configured or not.
        self.prompt_quiet_configured = None

+      if "Rollback Done" not in output:
+          raise SomeException(f"Automatic rollback to previous configuration failed:\n\n{output}")

        # Save config to startup
        self.device.save_config()
```
